examples/platform/linux/AppMain.cpp

```diff
@@ -124,10 +124,20 @@ void Cleanup()
     // TODO(16968): Lifecycle management of storage-using components like GroupDataProvider, etc
 }
 
+// TODO(#20664) REPL test will fail if signal SIGINT is not caught, temporarily keep following logic.
+
+// when the shell is enabled, don't intercept signals since it prevents the user from
+// using expected commands like CTRL-C to quit the application. (see issue #17845)
+// We should stop using signals for those faults, and move to a different notification
+// means, like a pipe. (see issue #19114)
+#if !defined(ENABLE_CHIP_SHELL)
 void StopSignalHandler(int signal)
 {
-    DeviceLayer::PlatformMgr().StopEventLoopTask();
+    ChipLogDetail(DeviceLayer, "Caught signal %d", signal);
```

Comment:
I know that it's only an example application, but I'd rather avoid using logging function in a signal handler. Logging function *might* call `printf()`, which **will call** `malloc()` when creating output string.